File: 백준/Gold/11404. 플로이드/플로이드.java
import java.io.*;
import java.util.*;

public class Main {
    static final int INF = 1000000000;

    public static void print(int[][] matrix) {
        for (int i = 0; i < matrix.length; i++) {
            for (int j = 0; j < matrix[i].length; j++) {
                System.out.print(matrix[i][j] + " ");
            }
            System.out.println();
        }
    }

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int N = Integer.parseInt(br.readLine());
        int M = Integer.parseInt(br.readLine());

        // matrix 초기화 및 입력
        int[][] dist = new int[N][N];

        for (int i = 0; i < N; i++) {
            for (int j = 0; j < N; j++) {
                if (i != j) {
                    dist[i][j] = INF;
                }
            }
        }

        for (int i = 0; i < M; i++) {
            StringTokenizer st = new StringTokenizer(br.readLine());
            int a = Integer.parseInt(st.nextToken());
            int b = Integer.parseInt(st.nextToken());
            int c = Integer.parseInt(st.nextToken());
            dist[a - 1][b - 1] = Math.min(dist[a - 1][b - 1], c);
        }

        // 플로이드 워셜
        for (int mid = 0; mid < N; mid++) {
            for (int start = 0; start < N; start++) {
                for (int end = 0; end < N; end++) {
                    if (dist[start][mid] == INF || dist[mid][end] == INF) {
                        continue;
                    }
                    dist[start][end] = Math.min(dist[start][end], dist[start][mid] + dist[mid][end]);
                }
            }
        }

        // 출력
        StringBuilder sb = new StringBuilder();
        for (int i = 0; i < N; i++) {
            for (int j = 0; j < N; j++) {
                if (dist[i][j] == INF) {
                    sb.append(0).append(' ');
                } else {
                    sb.append(dist[i][j]).append(' ');
                }
            }
            sb.append("\n");
        }

        System.out.print(sb);
    }
}

File: 백준/Gold/11404. 플로이드/플로이드.py
import sys
import copy
input = sys.stdin.readline
INF = int(1e9)

# 플로이드 워셜의 dist 행렬 출력하는 문제
N = int(input())
M = int(input())
adj_matrix = [[INF] * N for _ in range(N)]

for _ in range(M):
    a, b, c = map(int, input().split())
    adj_matrix[a - 1][b - 1] = min(adj_matrix[a - 1][b - 1], c)

# dist 행렬 초기화
dist = copy.deepcopy(adj_matrix)
for i in range(N):
    for j in range(N):
        if i == j:
            dist[i][j] = 0

# 플로이드 워셜
for mid in range(N):
    for start in range(N):
        for end in range(N):
            if dist[start][mid] == INF or dist[mid][end] == INF:
                continue
            dist[start][end] = min(dist[start][end], dist[start][mid] + dist[mid][end])

for row in dist:
    for elem in row:
        if elem == INF:
            print(0, end=" ")
        else:
            print(elem, end=" ")
    print()
